src/main/java/de/srendi/advancedperipherals/client/screens/KeyboardScreen.java

```diff
@@ -0,0 +1,218 @@
+package de.srendi.advancedperipherals.client.screens;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import dan200.computercraft.client.gui.ClientInputHandler;
+import dan200.computercraft.shared.computer.core.InputHandler;
+import de.srendi.advancedperipherals.client.screens.base.BaseScreen;
+import de.srendi.advancedperipherals.common.container.KeyboardContainer;
+import net.minecraft.SharedConstants;
+import net.minecraft.client.Minecraft;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.player.Inventory;
+import org.jetbrains.annotations.NotNull;
+import org.lwjgl.glfw.GLFW;
+
+import java.util.BitSet;
+
+/**
+ * A simple screen but without any rendering calls. Used to unlock the mouse so we can freely write stuff
+ * <p>
+ * Char/key logic stolen from CC's WidgetTerminal
+ */
+public class KeyboardScreen extends BaseScreen<KeyboardContainer> {
+
+    protected final InputHandler input;
+    private final BitSet keysDown = new BitSet(256);
+
+    private float terminateTimer = -1;
+    private float rebootTimer = -1;
+    private float shutdownTimer = -1;
+
+    private int lastMouseButton = -1;
+    private int lastMouseX = -1;
+    private int lastMouseY = -1;
+
+    public KeyboardScreen(KeyboardContainer screenContainer, Inventory inv, Component titleIn) {
+        super(screenContainer, inv, titleIn);
+        input = new ClientInputHandler(menu);
+    }
+
+    @Override
+    public void render(@NotNull PoseStack poseStack, int x, int y, float partialTicks) {
+        Minecraft minecraft = Minecraft.getInstance();
+        float scale = 2f;
+        int screenWidth = minecraft.getWindow().getGuiScaledWidth();
+        int screenHeight = minecraft.getWindow().getGuiScaledHeight();
+        // Make the text a bit smaller on small screens
+        if (screenWidth <= 1080)
+            scale = 1f;
+
+        poseStack.scale(scale, scale, 1);
+        String text = "Press ESC to close the Keyboard Screen.";
+        float textX = (screenWidth / 2f - minecraft.font.width(text) * scale / 2f) / scale;
+        minecraft.font.drawShadow(poseStack, text, textX, 1, 0xFFFFFF);
+
+        // Prevents JEI/REI/EMI from rendering. Maybe not the best way, but it works for now.
+        poseStack.scale(4f, 4f, 1);
+        minecraft.font.draw(poseStack, "", screenWidth, screenHeight, 0xFFFFFF);
+    }
+
+    @Override
+    protected void renderBg(@NotNull PoseStack matrixStack, float partialTicks, int x, int y) {
+    }
+
+    @Override
+    public void renderBackground(@NotNull PoseStack pPoseStack) {
+    }
+
+    @Override
+    public boolean charTyped(char ch, int modifiers) {
+        if (ch >= 32 && ch <= 126 || ch >= 160 && ch <= 255) { // printable chars in byte range
```

Comment:
ASCII char has a maximum value of `0x7f` (127) btw, not `0xff` or 255, above 0x7f will be considers as a two or more bytes UTF8

Reply:
Everything below chartTyped is borrowed from CCs screen itself, so I don't know why squid did what 😅